vspace/src/main/java/edu/asu/diging/vspace/web/staff/EditReferenceController.java

```diff
@@ -0,0 +1,104 @@
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.vspace.core.model.IReference;
+import edu.asu.diging.vspace.core.model.impl.Reference;
+import edu.asu.diging.vspace.core.references.ReferenceBlock;
+import edu.asu.diging.vspace.core.references.ReferenceDisplayDefault;
+import edu.asu.diging.vspace.core.references.ReferenceType;
+import edu.asu.diging.vspace.core.services.IReferenceManager;
+
+@Controller
+public class EditReferenceController {
+    
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private IReferenceManager referenceManager;
+    
+    @Autowired
+    private ReferenceDisplayDefault referenceDisplayProvider;
+    
+    @RequestMapping(value = "/staff/reference/{referenceId}/edit", method = RequestMethod.GET)
+    public String show(Model model, @PathVariable("referenceId") String referenceId, RedirectAttributes attributes) {
+        IReference reference = referenceManager.getReferenceById(referenceId);
+        model.addAttribute("referenceData", reference);
+        model.addAttribute("referenceTypes", ReferenceType.values());
+        return "staff/references/edit";
+    }
+    
+    /**
+     * This method allows updating References when a user is on the reference page accessed from 'References' option from side navbar.
+     */
+    @RequestMapping(value = "/staff/reference/{referenceId}/edit", method = RequestMethod.POST)
+    public String save(@ModelAttribute Reference refData, @PathVariable("referenceId") String referenceId, 
+            RedirectAttributes attributes) {
+        IReference reference = referenceManager.getReferenceById(referenceId);
+        if(reference!=null) {
+            reference.setTitle(refData.getTitle());
+            reference.setAuthor(refData.getAuthor());
+            reference.setYear(refData.getYear());
+            reference.setJournal(refData.getJournal());
+            reference.setUrl(refData.getUrl());
+            reference.setVolume(refData.getVolume());
+            reference.setIssue(refData.getIssue());
+            reference.setPages(refData.getPages());
+            reference.setEditors(refData.getEditors());
+            reference.setType(refData.getType());
+            reference.setNote(refData.getNote());
+            referenceManager.updateReference(reference);
```

Comment:
have you tried updating the reference object directly? Why the copying of the data?